bin/varnishd/mgt/mgt_child.c

```diff
@@ -248,16 +248,31 @@ child_listener(const struct vev *e, int what)
 static int __match_proto__(vev_cb_f)
 child_poker(const struct vev *e, int what)
 {
+	char * r = NULL;
+	unsigned status;
 
 	(void)e;
 	(void)what;
 	if (child_state != CH_RUNNING)
 		return (1);
 	if (child_pid < 0)
 		return (0);
-	if (!mgt_cli_askchild(NULL, NULL, "ping\n"))
-		return (0);
-	return (0);
+	if (!mgt_cli_askchild(&status, &r, "ping\n")) {
+		if (strncmp("PONG ", r, 5)) {
+			MGT_complain(C_ERR,
+			    "Unexpected reply from ping: %u %s",
+			    status, r);
+			MGT_Child_Cli_Fail();
+		}
+	}
+	else {
+		MGT_complain(C_ERR, "Did not get a 200 from ping: %u %s",
+		    status, r);
```

Comment:
Is it? You get status and r in both cases which covers not getting a pong
and something other than a 200 status.
Both are unexpected replies from the cli pov.
The end user will mostly not understand the difference or care about the
distinction.

On 15 Jul 2016 10:43 a.m., "hermunn" <notifications@github.com> wrote:

> In bin/varnishd/mgt/mgt_child.c
> <https://github.com/varnishcache/varnish-cache/pull/2019#discussion_r70947336>
> :
>
> >
> >  	(void)e;
> >  	(void)what;
> >  	if (child_state != CH_RUNNING)
> >  		return (1);
> >  	if (child_pid < 0)
> >  		return (0);
> > -	if (!mgt_cli_askchild(NULL, NULL, "ping\n"))
> > -		return (0);
> > -	return (0);
> > +	if (!mgt_cli_askchild(&status, &r, "ping\n")) {
> > +		if (strncmp("PONG ", r, 5)) {
> > +			MGT_complain(C_ERR,
> > +			    "Unexpected reply from ping: %u %s",
> > +			    status, r);
> > +			MGT_Child_Cli_Fail();
> > +		}
> > +	}
> > +	else {
> > +		MGT_complain(C_ERR, "Did not get a 200 from ping: %u %s",
> > +		    status, r);
>
> From a debugging perspective, I think the cases are different enough to
> produce slightly different errors. But I guess it is a matter of taste.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/varnishcache/varnish-cache/pull/2019/files/53d52408ce4cb6475fbda762a745d873344c7d16#r70947336>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOwefu0oNrXAIULJshXBZtTTwP9Hi4kks5qV1ZHgaJpZM4JNKGJ>
> .
>